var/spack/repos/builtin/packages/exodusii/package.py

```diff
@@ -46,34 +43,47 @@ class Exodusii(Package):
     homepage = "https://github.com/gsjaardema/seacas"
     url      = "https://github.com/gsjaardema/seacas/archive/master.zip"
 
-    version('2016-02-08',
-            git='https://github.com/gsjaardema/seacas.git', commit='dcf3529')
+    version('2016-08-09',
+            git='https://github.com/gsjaardema/seacas.git', commit='2ffeb1b')
+
+    variant('mpi', default=False, description='Enables MPI parallelism.')
 
-    depends_on('cmake@2.8.7:', type='build')
-    depends_on('hdf5~shared~mpi')
-    depends_on('netcdf~mpi')
+    depends_on('cmake@2.8.11:', type='build')
+    depends_on('mpi', when='+mpi')
 
-    patch('exodus-cmake.patch')
+    # https://github.com/gsjaardema/seacas/blob/master/NetCDF-Mapping.md
+    depends_on('netcdf~mpi maxdims=65536 maxvars=524288', when='~mpi')
+    depends_on('hdf5+shared~mpi', when='~mpi')
 
-    def patch(self):
-        ff = FileFilter('cmake-exodus')
+    depends_on('netcdf+mpi maxdims=65536 maxvars=524288', when='+mpi')
+    depends_on('hdf5+shared+mpi', when='+mpi')
```

Comment:
In terms of variant forwarding, I think your best bet at the moment is to not specify whether or not netcdf or hdf5 should be built with mpi. This is what I have been doing in other packages (see #1553 and #1909). #1553 in particular describes the potential problems that this can cause. For example, if Exodusii defaults to `~mpi` and NetCDF and/or HDF5 default to `+mpi`, you'll get the following error message:
```
$ spack install exodusii
==> Error: Invalid spec: 'netcdf@4.4.1%gcc@6.1.1+hdf4~mpi arch=linux-fedora24-x86_64'. Package netcdf requires variant +mpi, but spec asked for ~mpi
```
This can be solved by setting all of the packages to agree (always default to `+mpi`), but if anyone has a conflicting `packages.yaml`, it will break again. I would suggest using:
```python
depends_on('netcdf maxdims=65536 maxvars=524288')
depends_on('hdf5+shared')
```
and even setting Exodusii to default to `+mpi`.